runtime/src/cost_tracker.rs

```diff
@@ -180,8 +189,17 @@ impl CostTracker {
             return Err(CostTrackerError::WouldExceedAccountMaxLimit);
         }
 
-        if self.account_data_size.saturating_add(account_data_len) > MAX_ACCOUNT_DATA_LEN {
-            return Err(CostTrackerError::WouldExceedAccountDataMaxLimit);
+        // NOTE: Check if the total accounts data size is exceed *before* the block accounts data
+        // size.  This way, transactions are not unnecessarily retried.
```

Comment:
How is transactions are "not retried" if failed on this check? Is `WouldExceedAccountDataTotalLimit` handled differently in banking_stage?

Reply:
In `Bank:: load_and_execute_transactions()`, it decides which transactions to retry based on the TransactionError. By not including `WouldExceedMaxTotalAccountDataCostLimit` in the special handling, these transactions will not be retried. 

https://github.com/solana-labs/solana/blob/043086081fefad756700edfa9c68b917eadd6151/runtime/src/bank.rs#L3962-L3978

Reply:
inexplicit :) That'll work, but less obvious. Maybe create another issue to refactor that `bank`'s special error handling logic someday.

Reply:
New issue created! https://github.com/solana-labs/solana/issues/23619